target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -56,6 +56,9 @@ case "$FIRMWARE" in
 		caldata_extract "0:art" 0x1000 0x2f20
 		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_ascii 0:appsblenv ethaddr) +2)
 		;;
+	xiaomi,mi-router-hd)
+		caldata_extract "ART" 0x1000 0x2f20
```

Comment:
why no nvmem cell ?

Reply:
If these lines are deleted, then Wifi is not initialized:
```
ath10k 6.4 driver, optimized for CT firmware, probing pci device: 0x46.
 ath10k_pci 0000:01:00.0: enabling device (0140 -> 0142)
 ath10k_pci 0000:01:00.0: pci irq msi oper_irq_mode 2 irq_mode 0 reset_mode 0
 ath10k_pci 0000:01:00.0: qca9984/qca9994 hw1.0 target 0x01000000 chip_id 0x00000000 sub 168c:cafe
 ath10k_pci 0000:01:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
 ath10k_pci 0000:01:00.0: firmware ver 10.4b-ct-9984-fW-13-5ae337bb1 api 5 features mfp,peer-flow-ctrl,txstatus-noack,wmi-10.x-CT,ratemask-CT,regdump-CT,txrate-CT,flush-all-CT,pingpong-CT,ch-regs-CT,nop-CT,set-special-CT,tx-rc-CT,cust-stats-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT,wmi-bcn-rc-CT crc32 7ea63dc5
 ath10k_pci 0000:01:00.0: failed to fetch board data for bus=pci,vendor=168c,device=0046,subsystem-vendor=168c,subsystem-device=cafe from ath10k/QCA9984/hw1.0/board-2.bin
 ath10k_pci 0000:01:00.0: failed to fetch board-2.bin or board.bin from ath10k/QCA9984/hw1.0
 ath10k_pci 0000:01:00.0: failed to fetch board file: -12
 ath10k_pci 0000:01:00.0: could not probe fw (-12)

 ath10k 6.4 driver, optimized for CT firmware, probing pci device: 0x40.
 ath10k_pci 0001:01:00.0: enabling device (0140 -> 0142)
 ath10k_pci 0001:01:00.0: pci irq msi oper_irq_mode 2 irq_mode 0 reset_mode 0
 ath10k_pci 0001:01:00.0: qca99x0 hw2.0 target 0x01000000 chip_id 0x003b01ff sub 168c:0002
 ath10k_pci 0001:01:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
 ath10k_pci 0001:01:00.0: firmware ver 10.4b-ct-9980-fW-13-5ae337bb1 api 5 features mfp,peer-flow-ctrl,txstatus-noack,wmi-10.x-CT,ratemask-CT,regdump-CT,txrate-CT,flush-all-CT,pingpong-CT,ch-regs-CT,nop-CT,set-special-CT,tx-rc-CT,cust-stats-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT,wmi-bcn-rc-CT crc32 b36a12bf
 ath10k_pci 0001:01:00.0: failed to fetch board data for bus=pci,vendor=168c,device=0040,subsystem-vendor=168c,subsystem-device=0002 from ath10k/QCA99X0/hw2.0/board-2.bin
 ath10k_pci 0001:01:00.0: failed to fetch board-2.bin or board.bin from ath10k/QCA99X0/hw2.0
 ath10k_pci 0001:01:00.0: failed to fetch board file: -12
 ath10k_pci 0001:01:00.0: could not probe fw (-12)
```
And I can't understand this strange behavior.

Reply:
This is strange for one maybe the compatible is wrong. For the other maybe the soc is using a special subsystem kind and we need a special board-2.bin

Reply:
When building the firmware, a universal image `board-2.bin` for the Wi-Fi module is placed in directory ` /lib/firmware/$FIRMWARE`.
After the kernel is loaded, the `11-ath10k-caldata` script is launched
```
part=ART
mtd=$(find_mtd_chardev $part)
caldata_dd $mtd /lib/firmware/$FIRMWARE $count $offset
```
This script retrieves the actual `board-2.bin` image from the `ART` partition.
Image `board-2.bin` is placed in directory `/lib/firmware/$FIRMWARE`. And only then Wi-Fi modules are initialized.

I think option `nvmem-cell-names = "pre-calibration"` does something different.

But when using call `caldata_extract "ART"`, everything works fine!

Reply:
if correctly setup they do the same thing... so it's just that the nvmem-cell is not working... probably the compatible is wrong.

Reply:
Fixed